server/server.js

```diff
@@ -1,15 +1,141 @@
 const express = require("express");
+const cors = require("cors");
+const moment = require("moment");
+
+
+
 const app = express();
-const port = process.env.PORT || 5000;
 
+app.use(express.json());
+app.use(cors());
+
+const port = process.env.PORT || 5000;
 app.listen(port, () => console.log(`Listening on port ${port}`));
 
-// Store and retrieve your videos from here
-// If you want, you can copy "exampleresponse.json" into here to have some data to work with
-let videos = [];
+let videos = [{
+  "id": 523523,
+  "title": "Never Gonna Give You Up",
+  "url": "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
+  "rating": 23,
+  "time": "May 28th 2023, 8:47:44 pm"
+},
+{
+  "id": 523427,
+  "title": "The Coding Train",
+  "url": "https://www.youtube.com/watch?v=HerCR8bw_GE",
+  "rating": 230,
+  "time": "May 28th 2023, 8:47:44 pm"
+},
+{
+  "id": 82653,
+  "title": "Mac & Cheese | Basics with Babish",
+  "url": "https://www.youtube.com/watch?v=FUeyrEN14Rk",
+  "rating": 2111,
+  "time": "May 28th 2023, 8:47:44 pm"
+},
+{
+  "id": 858566,
+  "title": "Videos for Cats to Watch - 8 Hour Bird Bonanza",
+  "url": "https://www.youtube.com/watch?v=xbs7FT7dXYc",
+  "rating": 11,
+  "time": "May 28th 2023, 8:47:44 pm"
+},
+{
+  "id": 283634,
+  "title": "Learn Unity - Beginner's Game Development Course",
+  "url": "https://www.youtube.com/watch?v=gB1F9G0JXOo",
+  "rating": 211,
+  "time": "May 28th 2023, 8:47:44 pm"
+},
+{
+  "id": 562824,
+  "title": "Cracking Enigma in 2021 - Computerphile",
+  "url": "https://www.youtube.com/watch?v=RzWB5jL5RX0",
+  "rating": 111,
+  "time": "May 28th 2023, 8:47:44 pm"
+},
+{
+  "id": 442452,
+  "title": "Coding Adventure: Chess AI",
+  "url": "https://www.youtube.com/watch?v=U4ogK0MIzqk",
+  "rating": 671,
+  "time": "May 28th 2023, 8:47:44 pm"
+},
+{
+  "id": 536363,
+  "title": "Coding Adventure: Ant and Slime Simulations",
+  "url": "https://www.youtube.com/watch?v=X-iSQQgOd1A",
+  "rating": 76,
+  "time": "May 28th 2023, 8:47:44 pm"
+},
+{
+  "id": 323445,
+  "title": "Why the Tour de France is so brutal",
+  "url": "https://www.youtube.com/watch?v=ZacOS8NBK6U",
+  "rating": 73,
+  "time": "May 28th 2023, 8:47:44 pm"
+}];
+
 
 // GET "/"
 app.get("/", (req, res) => {
-  // Delete this line after you've confirmed your server is running
-  res.send({ express: "Your Backend Service is Running" });
+  res.status(200).json({ data: videos });
 });
+
+
+// // POST "/add"
+app.post("/add", (req, res) => {
+  const { title, url } = req.body;
+  const regex1 = /(^https:\/\/www\.youtube\.com\/watch\?v=)([0-9a-zA-Z=\-_+.,:;'?!@#|$%^&*<>()]){11}$/;
+  const regex2 = /(^https:\/\/www.youtube.com\/shorts\/)([0-9a-zA-Z=\-_+.,:;'?!@#|$%^*<>()]){11}$/;
```

Comment:
When using regex, it's good to add a comment explaining what the regex does because regex is horrible to read even when written well.